<?xml version="1.0"?>
<launch>
	<arg name="process_freq" default="30.0" />
	<arg name="is_eval" default="true" />
    <arg name="arcuo_parser" default="$(find kuam_aruco_tracking)/../../config/" />
    <arg name="big_marker_id" default="0" />
    <arg name="small_marker_id" default="1" />
    <arg name="big_marker_size_m" default="0.2" />
    <arg name="small_marker_size_m" default="0.06" />
    <arg name="filter_buf_size" default="20" />
    <arg name="compare_mode" default="false" />
    <arg name="marker_cnt_th" default="10" />

    <!-- Select only one data type -->
    <arg name="using_gazebo_data" default="true" />
    <arg name="using_logging_data" default="false" />
    <arg name="using_device_data" default="false" />

    <!-- Estimating method -->
    <!-- 0. Without filter -->
    <!-- 1. Moving average filter -->
    <!-- 2. Exponential moving average filter -->
    <arg name="estimating_method" default="0" />

    <arg name="noise_dist_th_m" default="1.0" />
    <arg name="noise_cnt_th" default="20" />

    <node pkg="kuam_aruco_tracking" type="aruco_tracking_node" name="aruco_tracking" cwd="node" output="screen">
        <param name="process_freq" value="$(arg process_freq)" />
        <param name="is_eval" value="$(arg is_eval)" />
        <param name="arcuo_parser" value="$(arg arcuo_parser)" />
        <param name="big_marker_id" value="$(arg big_marker_id)" />
        <param name="small_marker_id" value="$(arg small_marker_id)" />
        <param name="big_marker_size_m" value="$(arg big_marker_size_m)" />
        <param name="small_marker_size_m" value="$(arg small_marker_size_m)" />
        <param name="filter_buf_size" value="$(arg filter_buf_size)" />
        <param name="compare_mode" value="$(arg compare_mode)" />
        <param name="estimating_method" value="$(arg estimating_method)" />
        <param name="using_gazebo_data" value="$(arg using_gazebo_data)" />
        <param name="noise_dist_th_m" value="$(arg noise_dist_th_m)" />
        <param name="noise_cnt_th" value="$(arg noise_cnt_th)" />
        <param name="marker_cnt_th" value="$(arg marker_cnt_th)" />
    </node>

    <group if="$(arg using_device_data)">
        <include file="$(find kuam_usb_cam)/launch/usb_cam.launch"/>
    </group>
</launch>
